src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -632,11 +633,22 @@ internal override void ForceComplete(SourceLocation? locationOpt, CancellationTo
 
                             EnsureFieldDefinitionsNoted();
 
-                            // We've completed all members, so we're ready for the PointedAtManagedTypeChecks;
-                            // proceed to the next iteration.
-                            state.NotePartComplete(CompletionPart.MembersCompleted);
-                            break;
+
+                            if (state.NotePartComplete(CompletionPart.MembersCompletedChecksStarted))
+                            {
+                                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This cancellation token check seems dangerous. If we get here and hit a cancellation, it seems we'll never get back into this block because we've already set `CompletionPart.MembersCompletedChecksStarted`.

Reply:
I've move this before the `if`, you're right that it's dangerous.